website/src/contribute/committer-guide.md

```diff
@@ -41,6 +41,7 @@ Granularity of changes:
 * It is OK to keep separate commits for different logical pieces of the code, if they make reviewing and revisiting code easier
 * Making commits isolated is a good practice, authors should be able to relatively easily split the PR upon reviewer's request
 * Generally, every commit should compile and pass tests.
+* Avoid keeping in history formatting messages such as checktyle or spotless fixes. Squash them with previous commit and add formatting in the commit message.
```

Comment:
Typo "checktyle" -> "checkStyle"

`add formatting in the commit message.` - not sure that we need to add this into message of squashed commit or you mean something else?

Reply:
I meant when people squash their checkstyle/spotless commits into the previous one, they should add "and fix formatting" to the commit message into which they squash to explain why that many files are changed.

Reply:
If there are commits that are doing widespread formatting fixes not related to the main PR, they should probably be separate commits and not squashed.

Reply:
@robertwb I totally agree, even if such a PR would not be that often because, with spotlless plugin, all changes are merged with correct format. But if a PR that does vast formatting were to come, I think it is correct to have formatting commits. What I would like to avoid is to have minor formatting commits in feature PRs that bring nothing to commit history.

Reply:
I agree with @robertwb - all checkstyle/spotless formatting, that is not related to current PR, has to be done as a separate PR. In the same time, I believe that it even should not on be a case since we rely on the fact that all code in master already checkstyle/spotless compatible (otherwise, it wouldn't be merged). So, luckily, it's quite rare case imo.

Regarding adding the formatting comments into main commit - I don't think this information is useful since it's already **required** to merge only well-formatted code. So, it's a default behaviour and there is no need to ask committer to do/check this. On the other hand, squashing such commits has to be required for history clearness. 

So, I propose to modify the phrase a little bit in this way:
"Avoid keeping in history formatting messages such as checkstyle or spotless fixes. Squash such commits with previous one."

Reply:
+1